src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -313,9 +352,27 @@ private float addRow(
           RowBufferStats.getCombinedStats(statsMap.get(columnName), forkedColStats.getValue()));
```

Comment:
oh wow this looks like a thread unsafe operation here. RowBuffer.statsMap and tempStatsMap are both constructed as non-concurrent collections, but we don't prohibit concurrent insertRows on one channel!

Reply:
(nothig to fix for now we'll followup on this separately)